src/index.ts

```diff
@@ -19,66 +19,94 @@
-  'no-identical-conditions',
-  'no-identical-expressions',
-  'no-identical-functions',
-  'no-ignored-return',
-  'no-inverted-boolean-check',
-  'no-nested-switch',
-  'no-nested-template-literals',
-  'no-one-iteration-loop',
-  'no-redundant-boolean',
-  'no-redundant-jump',
-  'no-same-line-conditional',
-  'no-small-switch',
-  'no-unused-collection',
-  'no-use-of-empty-return-value',
-  'no-useless-catch',
-  'non-existent-operator',
-  'prefer-immediate-return',
-  'prefer-object-literal',
-  'prefer-single-boolean-return',
-  'prefer-while',
-];
-
-const sonarjsRuleModules: { [key: string]: any } = {};
+const rules: Record<string, TSESLint.RuleModule<any, any>> = {
+  'cognitive-complexity': cognitiveComplexity,
+  'elseif-without-else': elseifWithoutElse,
+  'max-switch-cases': maxSwitchCases,
+  'no-all-duplicated-branches': noAllDuplicatedBranches,
+  'no-collapsible-if': noCollapsibleIf,
+  'no-collection-size-mischeck': noCollectionSizeMischeck,
+  'no-duplicate-string': noDuplicateString,
+  'no-duplicated-branches': noDuplicatedBranches,
+  'no-element-overwrite': noElementOverwrite,
+  'no-empty-collection': noEmptyCollection,
+  'no-extra-arguments': noExtraArguments,
+  'no-gratuitous-expressions': noGratuitousExpressions,
+  'no-identical-conditions': noIdenticalConditions,
+  'no-identical-expressions': noIdenticalExpressions,
+  'no-identical-functions': noIdenticalFunctions,
+  'no-ignored-return': noIgnoredReturn,
+  'no-inverted-boolean-check': noInvertedBooleanCheck,
+  'no-nested-switch': noNestedSwitch,
+  'no-nested-template-literals': noNestedTemplateLiterals,
+  'no-one-iteration-loop': noOneIterationLoop,
+  'no-redundant-boolean': noRedundantBoolean,
+  'no-redundant-jump': noRedundantJump,
+  'no-same-line-conditional': noSameLineConditional,
+  'no-small-switch': noSmallSwitch,
+  'no-unused-collection': noUnusedCollection,
+  'no-use-of-empty-return-value': noUseOfEmptyReturnValue,
+  'no-useless-catch': noUselessCatch,
+  'non-existent-operator': nonExistentOperator,
+  'prefer-immediate-return': preferImmediateReturn,
+  'prefer-object-literal': preferObjectLiteral,
+  'prefer-single-boolean-return': preferSingleBooleanReturn,
+  'prefer-while': preferWhile,
+};
 
 const plugin = {
   configs: {},
   rules: {},
 };
 
-const recommendedLegacyConfig: TSESLint.Linter.Config = { plugins: ['sonarjs'], rules: {} };
+const recommendedLegacyConfig: TSESLint.Linter.ConfigType = { plugins: ['sonarjs'], rules: {} };
 const recommendedConfig: FlatConfig.Config = {
   plugins: {
     sonarjs: plugin,
   },
   rules: {},
 };
 
-sonarjsRules.forEach(rule => {
-  sonarjsRuleModules[rule] = require(`./rules/${rule}`);
-  const {
-    meta: {
-      docs: { recommended },
-    },
-  } = sonarjsRuleModules[rule];
-  recommendedConfig.rules![`sonarjs/${rule}`] = recommended === undefined ? 'off' : 'error';
-});
+for (const key in rules) {
+  const rule = rules[key];
+  const recommended = rule.meta.docs?.recommended;
```

Comment:
I think that, in this case, not destructuring is more readable than destructuring. Feel free to request a change if you think the destructured syntax of the original code is clearer:

```
const {
    meta: {
      docs: { recommended },
    },
  } = rule;
```